src/core_plugins/kibana/public/context/api/context.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+
+import { addComputedFields } from './utils/fields';
+import { getDocumentUid } from './utils/ids';
+import { createSuccessorsQuery } from './utils/queries.js';
+import { reverseQuerySort } from './utils/sorting';
+
+
+async function fetchSuccessors(es, indexPattern, anchorDocument, sort, size) {
+  const successorsQuery = prepareQuery(indexPattern, anchorDocument, sort, size);
+  const results = await performQuery(es, indexPattern, successorsQuery);
+  return results;
+}
+
+async function fetchPredecessors(es, indexPattern, anchorDocument, sort, size) {
+  const successorsQuery = prepareQuery(indexPattern, anchorDocument, sort, size);
+  const predecessorsQuery = reverseQuerySort(successorsQuery);
+  const reversedResults = await performQuery(es, indexPattern, predecessorsQuery);
```

Comment:
The sequence of operations is actually quite simple:

1. create the query `successorsQuery`
2. reverse the query (because we only have `search_after` and no `search_before` in Elasticsearch), creating the `predecessorsQuery`
3. execute the `predecessorsQuery`, getting back the results in reversed order, ergo `reversedResults`
4. reverse the results to get correctly ordered `results`